### PR TITLE
scope(daemon) Implement listing of processes

### LINKED ISSUE #72: Retrieve running processes List
As a developer I want to:
retrieve a list of all running processes.

Acceptance Criteria:
Retrieved data can be efficiently accessed by the UI

### PR DESCRIPTION
**Pull request by Mr-Kanister**
The list is generated by the daemon via the procfs crate. The following information is being transmitted via grpc: `pid`, `ppid`, `comm` and `state`. `state` is a single character and is mapped according to the following list: https://docs.rs/procfs/latest/procfs/process/enum.ProcState.html#variants

The implementation can be tested via the cli client: `cargo run --bin backend-daemon-cli`.
The daemon must be running: `cargo xtask daemon`.

Closes #72

### CODE REVIEW

File: rust/backend/daemon/src/server.rs [hunk 1]
```diff
@@ -48,20 +47,9 @@ impl Ziofa for ZiofaImpl {
...
-                },
-                Process {
-                    pid: 1741231,
-                    package: "com.example.org".to_string(),
-                },
-            ],
-        };
-        Ok(Response::new(response))
+        // TODO: Error handling
+        let processes = list_processes().unwrap();
```
Comment (BenediktZinn) [inline rust/backend/daemon/src/server.rs:51]: the .unwrap could be replaced with ? i think, the client can then decide what todo on error...

File: rust/shared/proto/ziofa.proto [hunk 2]
```diff
@@ -29,8 +29,10 @@ message ProcessList {
 }
 
 message Process {
-    uint32 pid = 1;
-    string package = 2;
+    int32 pid = 1;
```
Comment (BenediktZinn) [inline rust/shared/proto/ziofa.proto:32]: why int32 instead of uint32, can the pid be negative?

**Review by BenediktZinn — COMMENTED** [review]
Just some nitpicking, the rest looks good to me

File: rust/backend/daemon/src/server.rs (hunk 1 above)
Comment (fhilgers) [reply rust/backend/daemon/src/server.rs:51]: For that you need your own error type, that implements the `Into<Status>` trait. For an example you can look into `counter.rs`.

File: rust/shared/proto/ziofa.proto (hunk 2 above)
Comment (fhilgers) [reply rust/shared/proto/ziofa.proto:32]: It seems that linux itself uses i32 for pids so we should keep it as int32:

- https://internals.rust-lang.org/t/std-child-pid-type-compatibility/10473
- https://github.com/eminence/procfs/pull/250

Comment (Mr-Kanister) [reply rust/shared/proto/ziofa.proto:32]: Yes, it matches the procfs stat.pid type.

File: rust/backend/daemon/src/server.rs (hunk 1 above)
Comment (Mr-Kanister) [reply rust/backend/daemon/src/server.rs]: Done. Fits you?

**Review by fhilgers — APPROVED** [review]

**Pull request merged**